src/index.test.ts

```diff
@@ -63,6 +63,83 @@ describe('basic go usage', () => {
   });
 });
 
+describe('basic retryGo usage', () => {
+  const operations = {
+    successFn: () => new Promise((res) => res(2)),
+    errorFn: () => new Promise((_res, rej) => rej(new Error('Computer says no'))),
+  };
+
+  it('retries the specified number of times', async () => {
+    const retries = 3;
+    jest
+      .spyOn(operations, 'successFn')
+      .mockRejectedValueOnce(new Error('Error 1'))
+      .mockRejectedValueOnce(new Error('Error 2'));
+
+    const res = await retryGo(operations.successFn, { retries });
+    expect(operations.successFn).toHaveBeenCalledTimes(retries);
+    expect(res).toEqual(success(2));
+  });
+
+  it('retries and resolves after timing out', async () => {
+    jest.spyOn(operations, 'successFn').mockRejectedValueOnce(new Error('Operation timed out, retries left: 1'));
+
+    const res = await retryGo(operations.successFn, { retries: 2 });
+    expect(operations.successFn).toHaveBeenCalledTimes(2);
+    expect(res).toEqual(success(2));
+  });
+
+  it('retries and resolves unsuccessful asynchronous functions', async () => {
+    jest
+      .spyOn(operations, 'errorFn')
+      .mockRejectedValueOnce(new Error('Operation timed out, retries left: 1'))
+      .mockRejectedValueOnce(new Error('Operation timed out after final retry'));
+
+    const res = await retryGo(operations.errorFn);
+    expect(operations.errorFn).toHaveBeenCalledTimes(3);
+    expect(res).toEqual(fail(new Error('Computer says no')));
+  });
+});
+
+describe('basic retryTimeoutGo usage', () => {
+  const operations = {
+    successFn: () =>
+      new Promise((res) =>
+        setTimeout(() => {
+          res(2);
+        }, 20)
+      ),
+    errorFn: () => new Promise((_res, rej) => setTimeout(() => rej(new Error('Computer says no')), 20)),
+  };
+
+  it('retries and resolves timed out functions', async () => {
+    const res = await retryTimeoutGo(operations.successFn, { timeoutMs: 50, retries: 3 });
+    expect(res).toEqual(success(2));
+  });
+
+  it('retries and resolves successful asynchronous functions', async () => {
```

Comment:
Added one more test for the case where `go` is called from inside the timeout. Is this what is needed?